### PR TITLE
feat(leaveapp): Add getApproversList API to check multiple approvers' availability

### PR DESCRIPTION
**Pull request by Prathmesh-Copods**
- Rename getApproverAvailability to getApproversList for better clarity
- Modify API to return list of approvers with their details and availability
- Add optional filtering by role and department
- Combine ApproverListSerializer data with availability status
- Include approver's leave dates when unavailable
- Maintain existing date validation and availability checking logic

Input params:
- role (optional)
- department (optional)
- requested_leave_start_date (required)
- requested_leave_end_date (required)

Response format:
Returns array of objects containing user details with availability status

approver list: 
image

### CODE REVIEW

File: LeaveTrackingApp/views/leaves_views.py [hunk 1]
```diff
@@ -629,4 +630,65 @@ def withdrawLeave(request, id):
...
+    -   Check approver availability during escalation period
+    -   Return availability status with leave details if unavailable
+    -   Handle errors and return appropriate error responses
+    """
+    if request.method == 'GET':
+        try:
+            approver_id = request.GET.get('approver_id')
+            start_date = request.GET.get('requested_leave_start_date')
+            end_date = request.GET.get('requested_leave_end_date')
+
```
Comment (chandani-copods) [inline LeaveTrackingApp/views/leaves_views.py:653]: Dont get data from params, get it from payload.

**Comment by chandani-copods** [comment]
Resolve conflict.

File: LeaveTrackingApp/views/leaves_views.py [hunk 2]
```diff
@@ -629,4 +630,65 @@ def withdrawLeave(request, id):
                 else:
                     return JsonResponse({'msg': 'Withdrawn Successfully'}, status=200)
         except Exception as e:
-            return JsonResponse({'error': str(e)}, status=500)
\ No newline at end of file
+            return JsonResponse({'error': str(e)}, status=500)
+@csrf_exempt
+@user_is_authorized
+def getApproverAvaibility(request):
```
Comment (chandani-copods) [inline LeaveTrackingApp/views/leaves_views.py:636]: correct the spelling of availability.

File: LeaveTrackingApp/utils.py [hunk 3]
```diff
@@ -947,4 +947,69 @@ def is_leave_valid(leave_data):
...
+    escalation_start_date = end_date + timedelta(days=1)
+    escalation_end_date = escalation_start_date + timedelta(days=escalation_days - 1)
+    
+    approver_leaves = Leave.objects.filter(
+        Q(user_id=approver_id) &
+        Q(status='A') &
+        Q(start_date__lte=escalation_end_date) &
+        Q(end_date__gte=escalation_start_date)
+    ).order_by('start_date').first()
+
```
Comment (chandani-copods) [inline LeaveTrackingApp/utils.py:1005]: while testing, i am not getting any approver_leaves even if i have leave days.

Comment (Prathmesh-Copods) [reply LeaveTrackingApp/utils.py:1005]: Tested, its working on dev db and locally

File: LeaveTrackingApp/views/leaves_views.py (hunk 2 above)
Comment (Prathmesh-Copods) [reply LeaveTrackingApp/views/leaves_views.py]: fixed

File: LeaveTrackingApp/utils.py [hunk 4]
```diff
@@ -1210,6 +1211,73 @@ def is_leave_valid(leave_data):
...
+    """
+    if isinstance(start_date, str):
+        start_date = datetime.strptime(start_date, "%Y-%m-%d").date()
+    if isinstance(end_date, str):
+        end_date = datetime.strptime(end_date, "%Y-%m-%d").date()
+
+    leave_days = get_leave_days(start_date, end_date)
+    escalation_days = calculate_escalation_period(leave_days)
+    
+    escalation_start_date = end_date + timedelta(days=1)
```
Comment (chandani-copods) [inline LeaveTrackingApp/utils.py:1259]: logically in this function, Escalation_start_date should be leave creation date, right?

File: LeaveTrackingApp/utils.py [hunk 5]
```diff
@@ -1210,6 +1211,73 @@ def is_leave_valid(leave_data):
...
+    escalation_end_date = escalation_start_date + timedelta(days=escalation_days - 1)
+    
+    approver_leaves = Leave.objects.filter(
+        Q(user_id=approver_id) &
+        Q(status='A') &
+        Q(start_date__lte=escalation_end_date) &
+        Q(end_date__gte=escalation_start_date)
+    ).order_by('start_date').first()
+
```
Comment (chandani-copods) [inline LeaveTrackingApp/utils.py:1268]: here, i have leave requested of 25,26 and 27 nov that i created on 24 nov from different id but i am not getting any leave days here. According to logic if i am creating leave today on 26 nov then i should get the data of approver who is on leave on 26 and 27. but i am getting none here.

Comment (Prathmesh-Copods) [reply LeaveTrackingApp/utils.py:1268]: Ok, checking it with dev DB

File: LeaveTrackingApp/utils.py (hunk 4 above)
Comment (Prathmesh-Copods) [reply LeaveTrackingApp/utils.py]: yes fixed

**Pull request merged**